airflow/providers/apache/spark/hooks/spark_sql.py

```diff
@@ -152,8 +160,15 @@ def _prepare_command(self, cmd: Union[str, List[str]]) -> List[str]:
                 connection_cmd += ["-f", sql]
             else:
                 connection_cmd += ["-e", sql]
-        if self._master:
-            connection_cmd += ["--master", self._master]
+        if len(self._conn.host) > 0:
```

Comment:
```suggestion
        if self._conn.host:
```

is a more normal way of doing this -- we don't actually care about the length, so lets not compute it

Reply:
If `self._master` is specified, that should be used over anything in the connection, otherwise you are breaking existing behaviuor.

Reply:
Okay